apps/meteor/app/livechat/server/api/v1/room.ts

```diff
@@ -352,7 +352,12 @@ API.v1.addRoute(
 
 			const chatForwardedResult = await Livechat.transfer(room, guest, transferData);
 
-			return chatForwardedResult ? API.v1.success() : API.v1.failure();
+			if (!chatForwardedResult) {
+				const { departmentId } = transferData;
+				return API.v1.failure(departmentId ? 'error-no-agents-online-in-department' : 'error-forwarding-chat');
+			}
+
+			return API.v1.success();
```

Comment:
The rest of the changes look good! I'm concerned about falsely reporting in-correct error messages here. Ideally, this error should be thrown from the `Livechat.transfer` method when we check if no agents are online within the department.

Reply:
I'm checking if we can avoid this logic and move it inside the transfer method. Will keep you posted about updates here

Reply:
This should be addressed with my latest commit